2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh

```diff
@@ -201,84 +201,140 @@ kafka_create_alias_environment_variables() {
 kafka_validate() {
     debug "Validating settings in KAFKA_* env vars..."
     local error_code=0
+    local internal_port
+    local client_port
 
     # Auxiliary functions
     print_validation_error() {
         error "$1"
         error_code=1
     }
-
-    local validate_port_args=()
-    ! am_i_root && validate_port_args+=("-unprivileged")
-    for var in "KAFKA_PORT_NUMBER"; do
-        if ! err=$(validate_port "${validate_port_args[@]}" "${!var}"); then
-            print_validation_error "An invalid port was specified in the environment variable $var: $err"
+    check_allowed_listener_port() {
+        local -r total="$#"
+        for i in $(seq 1 "$((total - 1))"); do
+            for j in $(seq "$((i + 1))" "$total"); do
+                if (( "${!i}" == "${!j}" )); then
+                    print_validation_error "There are listeners bound to the same port"
+                fi
+            done
+        done
+    }
+    check_conflicting_listener_ports() {
+        local validate_port_args=()
+        ! am_i_root && validate_port_args+=("-unprivileged")
+        if ! err=$(validate_port "${validate_port_args[@]}" "$1"); then
+            print_validation_error "An invalid port was specified in the environment variable KAFKA_CFG_LISTENERS: $err"
         fi
-    done
+    }
+
+    if [[ ${KAFKA_CFG_LISTENERS:-} =~ INTERNAL:\/\/:([0-9]*) ]]; then
+        internal_port="${BASH_REMATCH[1]}"
+        check_allowed_listener_port "$internal_port"
+    fi
+    if [[ ${KAFKA_CFG_LISTENERS:-} =~ CLIENT:\/\/:([0-9]*) ]]; then
+        client_port="${BASH_REMATCH[1]}"
+        check_allowed_listener_port "$client_port"
+    fi
+    [[ -n ${internal_port:-} && -n ${client_port:-} ]] && check_conflicting_listener_ports "$internal_port" "$client_port"
```

Comment:
For consistency, we should use quotes to contain environment variables:

```suggestion
    if [[ "${KAFKA_CFG_LISTENERS:-}" =~ INTERNAL:\/\/:([0-9]*) ]]; then
        internal_port="${BASH_REMATCH[1]}"
        check_allowed_listener_port "$internal_port"
    fi
    if [[ "${KAFKA_CFG_LISTENERS:-}" =~ CLIENT:\/\/:([0-9]*) ]]; then
        client_port="${BASH_REMATCH[1]}"
        check_allowed_listener_port "$client_port"
    fi
    [[ -n "${internal_port:-}" && -n "${client_port:-}" ]] && check_conflicting_listener_ports "$internal_port" "$client_port"
```